python/dcr_template.py

```diff
@@ -298,175 +618,225 @@ def load_model(self, model_repository=None, band_name='g', **kwargs):
             n_step = int(np.ceil((bandpass.wavelen_max - bandpass.wavelen_min) / bandpass.wavelen_step))
         self.n_step = n_step
         self.bandpass = bandpass
         self.photoParams = PhotometricParameters(exptime=exposure_time, nexp=1, platescale=self.pixel_scale,
                                                  bandpass=band_name)
-        dcr_test = _dcr_generator(bandpass, pixel_scale=self.pixel_scale, elevation=elevation_min, azimuth=0.)
+
+        elevation_min = np.min(self.elevation_arr) - 5.  # Calculate slightly worse DCR than maximum.
+        dcr_test = DcrModel._dcr_generator(bandpass, pixel_scale=self.pixel_scale,
+                                           elevation=elevation_min, azimuth=0.)
         self.dcr_max = int(np.ceil(np.max(dcr_test.next())) + 1)
         if kernel_size is None:
-            self.kernel_size = 2*self.dcr_max + 1
+            kernel_size = 2*self.dcr_max + 1
         else:
-            self.kernel_size = kernel_size
-        self.use_psf = use_psf
-        self.use_fft = use_fft
-        self.debug = debug_mode
-        self.regularize = self._build_regularization(x_size=self.kernel_size,
-                                                     y_size=self.kernel_size, n_step=self.n_step)
+            kernel_size = 2*(kernel_size//2) + 1  # Ensure kernel is odd
+        self.kernel_size = int(kernel_size)
+        self.use_psf = bool(use_psf)
+        self.debug = bool(debug_mode)
+        self.regularize = self._build_regularization(x_size=self.kernel_size, y_size=self.kernel_size,
+                                                     n_step=self.n_step, frequency_regularization=True)
         self._calc_psf_model()
 
     @staticmethod
-    def _build_regularization(x_size=None, y_size=None, n_step=None):
-        """
-        Regularization adapted from Nate Lust's DCR Demo iPython notebook.
+    def _build_regularization(x_size=None, y_size=None, n_step=None, spatial_regularization=False,
+                              frequency_regularization=False, frequency_second_regularization=False):
+        """Regularization adapted from Nate Lust's DCR Demo iPython notebook.
+
         Calculate a difference matrix for regularization as if each wavelength were a pixel, then scale
         the difference matrix to the size of the number of pixels times number of wavelengths
+        @param x_size: width of the kernel, in pixels.
+        @param y_size: height of the kernel, in pixels.
+        @param n_step: Number of sub-filter wavelength planes.
+        @param spatial_regularization: Flag, set to True to include spatial regularization (not recommended)
+        @param frequency_regularization: Flag, set to True to regularize using frequency smoothness
+        @param frequency_second_regularization: Flag, set to True to regularize using smoothness of the
+               derivative of the frequency.
```

Comment:
What does this return? (Please check for other methods that return undocumented values, as well).